test/models/user_test.rb

```diff
@@ -320,6 +320,50 @@ def setup_user(operation)
     assert_includes record.errors.keys, :admin
   end
 
+  test "audit of password change should be saved only once, second time audited changes should not contain password_changed" do
+    user = FactoryGirl.create(:user)
+    user = User.find_by_id(user.id) #to clear the value of user.password
+    as_admin do
+      user.password = "newpassword"
+      assert_valid user
+      assert user.password_changed_changed?
+      assert user.password_changed
+      assert_includes user.changed, "password_changed"
+      assert user.save
+    end
+    user = User.find_by_id(user.id) #to clear the value of user.password
```

Comment:
This is this meant to test the `after_save`, isn't it? Replacing the entire user object will also reset the value, so it doesn't test the callback.